lib/install/planbuilder.go

```diff
@@ -324,6 +324,48 @@ func (b *PlanBuilder) AddWaitPhase(plan *storage.OperationPlan) {
 	})
 }
 
+// AddRegisterNodesPhase appends phases to register each node with the cluster
+func (b *PlanBuilder) AddRegisterNodesPhase(plan *storage.OperationPlan) error {
+	var newPhases []storage.OperationPhase
+
+	for i, node := range b.Masters {
```

Comment:
Could be:
```go
for i, node := range append(b.Masters, b.Nodes...) {
```

Reply:
So this doesn't work in this case, atleast without some other tweaks. I think it has to do with the Server object in OperatePhaseData being a pointer, this seems to cause the pointer to point to the temporary node location, and then get changed by subsequent loop iterations. By keeping this as separate loops it's just pointing to the prebuilt structure of masters and servers.

I'm not sure if I'm missing something, but I had tried this change and ran into that glitch.